architecture.md

```diff
@@ -0,0 +1,103 @@
+# Design overview - Architecture
+
+This issue should serve as an initial guidance and quick glance overview for Peribolos as a Service
+
+```mermaid
+flowchart LR
+    gh[GitHub] -- Webhook events trigger --> c[GitHub App controller]
```

Comment:
@harshad16  Hm.. how would you address that? The data really flows from GitHub servers as github webhook events to github app controller on our side. The "github app" is not generating the events, it's receiving them.

Reply:
That is correct, however, the GitHub webhooks are visible only at the GitHub app webhook console.
I would have made it
`gh-app[Github App] -- Webhook events trigger --> c[GitHub App controller]`

just to have a clear understanding from the user's side, the webhook flow from GitHub app (even though its GitHub servers) instead of thing the webhook is flowing out directly from Github repo. 
this is just my suggestion. if other feel it alright, then we are good. :+1: